protocol/protocol.proto

```diff
@@ -659,7 +659,10 @@ message Minipool {
 }
 
 message Tags {
-    map<string, string> tags = 1;
+    MessageInteractionType message_interaction_type = 1;
+    GroupMentionType group_mention_type = 2;
```

Comment:
my take is that @role need to be resolved at the client first and converted to user arrays?

Reply:
@person is represented in the mentioned_user_addresses
@role - roles are complicated, it's not really feasible for the client to calculate roles because they have to do a nasty lookup on every wallet for every user in a channel... I think roles will end up getting cached in the stream-metadata service. I think i will remove them from the payloads.proto file just to force us to think about it in the future
@channel - that's here, it's currently the only supported GroupMentionType

Ultimately you are right though - this should be a `repeated GroupMentionType group_mention_type`, i'll fix it.

Reply:
if we need role mentions and we don't implement them as regular mentions, we can add `repeated RoleMention role_mentions = 5` to capture the role id and anything else needed to broadcast to the role.

Reply:
if the sender tags multiple @channel, @person, -> the SW should only show the most specific tag. NS should send all the tags, in my opinion, and let the client decide based on its UX design.

@role is very tricky, especially roles with token entitlements. Users can sell or buy assets at any time, and may be expecting or not expecting notifications.